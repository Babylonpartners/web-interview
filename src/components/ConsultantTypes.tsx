import * as React from 'react'
import { headerStyles, activeButtonStyles } from '../styles/styles';
import { Responsive } from 'semantic-ui-react';

interface Props {
    handleChange: any
    selectedConsultantType: any
    availableSlots: any
}

export class ConsultantTypes extends React.PureComponent <Props> {
    render() {

        let consultantArray = []
        let slotData = this.props.availableSlots
        let consultantData = this.props.availableSlots
    
        // calculate unique consultant types
    
        for (let i = 0, len=consultantData.length ; i < len; i++) {
          let consultantType = slotData[i]['consultantType' as any]
          consultantArray.push(consultantType)
        }
        let merged = [].concat.apply([], consultantArray as any)
        const uniqueConsultantTypes = [...new Set(merged) as any]

        if(uniqueConsultantTypes.length < 1) return <p data-testid="consultantTypesLoading">Loading...</p>
        return (

            <div data-testid="consultantTypesComponent">
            <br />
            <br />

            <Responsive maxWidth={640}>
            <strong
                style={headerStyles}
            >
                Consultant Type
            </strong>
            <br />
            <br />
            <>
            
            {uniqueConsultantTypes.map((consultantType: any, i: any) => (
              <React.Fragment
                key={i}
              >
              {this.props.selectedConsultantType === consultantType
              ?
              <button
                style={activeButtonStyles}
                value={consultantType}
                className="mobile-button"
                onClick={this.props.handleChange}
              >
              {consultantType.charAt(0).toUpperCase() + consultantType.slice(1)}  
              </button>
              :
              <button
                style={activeButtonStyles}
                value={consultantType}
                className="mobile-white-button"
                onClick={this.props.handleChange}
              >
                {consultantType.charAt(0).toUpperCase() + consultantType.slice(1)}  
              </button>
              }
              </React.Fragment>
            ))}
            </>
            </Responsive>





            <Responsive minWidth={641}>
            <strong
                style={headerStyles}
            >
                Consultant Type
            </strong>
            <br />
            <br />
            <>
            {uniqueConsultantTypes.map((consultantType: any, i: any) => (
              <React.Fragment
                key={i}
              >
              {this.props.selectedConsultantType === consultantType
              ?
              <button
                style={activeButtonStyles}
                value={consultantType}
                className="button"
                onClick={this.props.handleChange}
              >
              {consultantType.charAt(0).toUpperCase() + consultantType.slice(1)}  
              </button>
              :
              <button
                style={activeButtonStyles}
                value={consultantType}
                className="white-button"
                onClick={this.props.handleChange}
              >
                {consultantType.charAt(0).toUpperCase() + consultantType.slice(1)}  
              </button>
              }
              </React.Fragment>
            ))}
            </>
            </Responsive>

            </div>

        );
}}