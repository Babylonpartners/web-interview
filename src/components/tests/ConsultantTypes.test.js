import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { ConsultantTypes } from '../ConsultantTypes'

afterEach(() => {
  cleanup()
  console.error.mockClear()
})

console.error = jest.fn()

const slots = [
  {
    id: 1,
    consultantType: ['gp'],
    appointmentType: ['audio', 'video'],
    time: '2019-11-27T10:11:00.000Z',
  },
  {
    id: 2,
    consultantType: ['specialist', 'gp'],
    appointmentType: ['audio', 'video'],
    time: '2019-12-01T14:16:30.000Z',
  },
]

test('<ConsultantTypes /> slot data NOT recieved', () => {
  const { getByTestId, queryByTestId, container } = render(
    <ConsultantTypes
      selectedAppointmentType
      selectedConsultantType
      handleChange
      availableSlots
    />
  )
  expect(container.firstChild).toMatchSnapshot()
  expect(getByTestId('consultantTypesLoading')).toBeTruthy()
  expect(queryByTestId('consultantTypesComponent')).toBeFalsy()
})

test('<ConsultantTypes /> slot data NOT recieved', () => {
  const { getByTestId, queryByTestId } = render(
    <ConsultantTypes
      selectedAppointmentType
      selectedConsultantType
      handleChange
      availableSlots={slots}
    />
  )

  expect(queryByTestId('consultantTypesLoading')).toBeFalsy()
  expect(getByTestId('consultantTypesComponent')).toBeTruthy()
})
