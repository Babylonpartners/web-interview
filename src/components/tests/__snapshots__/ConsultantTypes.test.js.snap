// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConsultantTypes /> slot data NOT recieved 1`] = `
<p
  data-testid="consultantTypesLoading"
>
  Loading...
</p>
`;
